sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -169,6 +169,39 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase {
       summary = getSummary(context))
   }
 
+  def invalidInputInConversionError(
+      to: DataType,
+      s: UTF8String,
+      fmt: UTF8String,
+      hint: String = "",
+      context: SQLQueryContext): SparkIllegalArgumentException = {
+    val alternative = if (hint.nonEmpty) {
+      s" Use '$hint' to tolerate malformed input and return NULL instead."
+    } else ""
+    new SparkIllegalArgumentException(
```

Comment:
Note: with error class, we don't need different java exception classes and can always use `SparkException`. We keep these special exception classes only for backward compatibility.

Reply:
`SparkException` does not support `QueryContext` yet, will need a separate PR to fix that

Reply:
@MaxGekk can you help to fix it?

Reply:
I opened the JIRA for this https://issues.apache.org/jira/browse/SPARK-40174